src/service-override/storage.ts

```diff
@@ -0,0 +1,144 @@
+import { Emitter, Event } from 'vs/base/common/event'
+import { IStorage, IStorageDatabase, IStorageItemsChangeEvent, IUpdateRequest, Storage } from 'vs/base/parts/storage/common/storage'
+import { IEditorOverrideServices } from 'vs/editor/standalone/browser/standaloneServices'
+import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors'
+import { AbstractStorageService, IStorageService, StorageScope as VSStorageScope } from 'vs/platform/storage/common/storage'
+import { IUserDataProfile } from 'vs/platform/userDataProfile/common/userDataProfile'
+import { IAnyWorkspaceIdentifier } from 'vs/platform/workspace/common/workspace'
+
+export enum StorageScope {
+  APPLICATION = 'application',
+  PROFILE = 'profile',
+  WORKSPACE = 'user',
+}
+
+export interface IStorageProvider {
+  read (scope: StorageScope): Map<string, string> | undefined
+  write (scope: StorageScope, data: Map<string, string>): Promise<void>
```

Comment:
Now that I'm rereading the code, I think the return type of `read` and second arg of `write` should be something that can already be serialized easily by provider w/o the need to deal with `Map`s.

Reply:
> So will you change something here at the end?

No, in the end I don't really think it's that big of a deal to work with maps for the implementer of provider.